sdks/python/apache_beam/transforms/display.py

```diff
@@ -0,0 +1,253 @@
+        self.items.append(element)
+        continue
+
+      # If it's not a HasDisplayData element,
+      # nor a dictionary, then it's a simple value
+      self.items.append(
+          DisplayDataItem(element,
+                          namespace=self.namespace,
+                          key=key))
+
+  def output(self):
+    """ Returns the JSON-API list of display data items to send to the runner.
+    """
+    return [item.get_dict() for item in self.items]
+
+  @classmethod
+  def create_from(cls, has_display_data):
+    """ Creates DisplayData from a HasDisplayData instance.
+
+    Returns:
+      A DisplayData instance with populated items.
+
+    Raises:
+      ValueError: If the has_display_data argument is not an instance of
+        HasDisplayData.
+    """
+    if not isinstance(has_display_data, HasDisplayData):
+      raise ValueError('Element of class {}.{} does not subclass HasDisplayData'
+                       .format(has_display_data.__module__,
+                               has_display_data.__class__.__name__))
+    return cls(has_display_data._namespace(), has_display_data.display_data())
+
+
+class DisplayDataItem(object):
+  """ A DisplayDataItem represents a unit of static display data.
+
+  Each item is identified by a key and the namespace of the component the
+  display item belongs to.
+  """
+  typeDict = {str:'STRING',
+              int:'INTEGER',
+              float:'FLOAT',
+              timedelta:'DURATION',
+              datetime:'TIMESTAMP'}
+
+  def __init__(self, value, url=None, label=None,
+               namespace=None, key=None, shortValue=None):
+    self.namespace = namespace
+    self.key = key
+    self.type = self._get_value_type(value)
+    self.shortValue = (shortValue if shortValue is not None else
+                       self._get_short_value(value, self.type))
+    self.value = value
+    self.url = url
+    self.label = label
+
+  def is_valid(self):
+    """ Checks that all the necessary fields of the DisplayDataItem are
+    filled in. It checks that neither key, namespace, value or type are None.
+
+    Raises:
+      ValueError: If the item does not have a key, namespace, value or type.
+    """
+    if self.key is None:
+      raise ValueError('Key must not be None')
+    if self.namespace is None:
+      raise ValueError('Namespace must not be None')
+    if self.value is None:
+      raise ValueError('Value must not be None')
+    if self.type is None:
+      raise ValueError('Value {} is of an unsupported type.'.format(self.value))
+
+  def get_dict(self):
+    """ Returns the internal-API dictionary representing the DisplayDataItem.
+
+    Returns:
+      A dictionary. The internal-API dictionary representing the
+      DisplayDataItem
+
+    Raises:
+     ValueError: if the item is not valid.
+    """
+    self.is_valid()
+
+    res = {'key': self.key,
+           'namespace': self.namespace,
+           'type': self.type if self.type != 'JAVA_CLASS' else 'STRING'}
```

Comment:
I see that you are now converting `JAVA_CLASS` types to string, but why have `JAVA_CLASS` in the first place? Should this be `CLASS`?